File: Java/sga/model/Alumno.java
package sga.model;

import java.util.ArrayList;
import java.util.List;

public class Alumno extends Persona {
    private List<Float> notas;
    private ProgramaAcademico programa;

    public Alumno(String cedula, String nombre, String correo, ProgramaAcademico programa) {
        super(cedula, nombre, correo);
        this.programa = programa;
        this.notas = new ArrayList<>();
    }

    public boolean agregarNota(float nota) {
        if (notas.size() < 3) {
            notas.add(nota);
            return true;
        }
        return false;
    }

    public Float deshacerNota() {
        if (!notas.isEmpty()) {
            return notas.remove(notas.size() - 1);
        }
        return null;
    }

    public List<Float> getNotas() {
        return notas;
    }

    public ProgramaAcademico getPrograma() {
        return programa;
    }

    public float obtenerPromedio() {
        if (notas.isEmpty()) {
            return 0.0f;
        }
        float suma = 0;
        for (float n : notas) {
            suma += n;
        }
        return suma / notas.size();
    }

    public boolean estaAprobado() {
        return programa.evaluarAprobacion(notas);
    }

    @Override
    public String mostrarInformacion() {
        String estatus = estaAprobado() ? "APROBADO" : "REPROBADO";
        String notasStr = notas.isEmpty() ? "Sin notas" : notas.toString();
        return String.format("[Alumno] ID: %s | Nombre: %s | Correo: %s | Prog: %s | Notas: %s | Prom: %.1f | Estatus: %s",
                cedula, nombre, correo, programa.getNombrePrograma(), notasStr, obtenerPromedio(), estatus);
    }
}

File: C++/Alumno.cpp
#include "Alumno.h"
#include <sstream>
#include <iomanip>

Alumno::Alumno(const std::string& cedula, const std::string& nombre, 
               const std::string& correo, ProgramaAcademico* programa)
    : Persona(cedula, nombre, correo), programa(programa) {}

Alumno::~Alumno() {
    // No eliminamos programa aquí porque lo gestiona SistemaGestionAcademica
}

bool Alumno::agregarNota(float nota) {
    if (notas.size() < 3) {
        notas.push_back(nota);
        return true;
    }
    return false;
}

float Alumno::deshacerNota() {
    if (!notas.empty()) {
        float ultima = notas.back();
        notas.pop_back();
        return ultima;
    }
    return -1.0f;
}

std::vector<float> Alumno::getNotas() const {
    return notas;
}

ProgramaAcademico* Alumno::getPrograma() const {
    return programa;
}

float Alumno::obtenerPromedio() const {
    if (notas.empty()) {
        return 0.0f;
    }
    float suma = 0;
    for (float n : notas) {
        suma += n;
    }
    return suma / notas.size();
}

bool Alumno::estaAprobado() const {
    return programa->evaluarAprobacion(notas);
}

std::string Alumno::mostrarInformacion() const {
    std::ostringstream oss;
    std::string estatus = estaAprobado() ? "APROBADO" : "REPROBADO";
    
    oss << "[Alumno] ID: " << cedula 
        << " | Nombre: " << nombre 
        << " | Correo: " << correo
        << " | Prog: " << programa->getNombrePrograma()
        << " | Prom: " << std::fixed << std::setprecision(1) << obtenerPromedio()
        << " | Estatus: " << estatus;
    
    return oss.str();
}